b2luigi/batch/processes/gbasf2.py

```diff
@@ -585,12 +612,33 @@ def _download_dataset(self):
             tmp_output_dir_path = f"{output_dir_path}.partial"
             os.makedirs(tmp_output_dir_path, exist_ok=True)
 
-            ds_get_command = shlex.split(f"gb2_ds_get --force {dataset_query_string}")
-            print("Downloading dataset with command ", " ".join(ds_get_command))
+            # Need a file to repeat download for FAILED ones only
+            monitoring_failed_downloads_file = os.path.join(tmp_output_dir_path, "failed_files.txt")
+
+            # In case of first download, this file does not exist
+            if not os.path.exists(monitoring_failed_downloads_file):
+
+                ds_get_command = shlex.split(f"gb2_ds_get --force {dataset_query_string}")
+                print("Downloading dataset with command ", " ".join(ds_get_command))
+
+            # Any further time is based on the list of files from failed downloads
+            else:
+                ds_get_command = shlex.split(f"gb2_ds_get --force --input_dslist {monitoring_failed_downloads_file}")
+                print("Downloading remaining files from dataset with command ", " ".join(ds_get_command))
+
             stdout = run_with_gbasf2(ds_get_command, cwd=tmp_output_dir_path, capture_output=True).stdout
             print(stdout)
             if "No file found" in stdout:
                 raise RuntimeError(f"No output data for gbasf2 project {self.gbasf2_project_name} found.")
+
+            failed_files = stdout.split('Failed files:')[-1].\
+                split("Files with duplicated jobID, not downloaded:")[0].strip().split('\n')
+            if len(failed_files) > 0:
+                with open(monitoring_failed_downloads_file, 'w') as ffs:
+                    ffs.write("\n".join(failed_files))
+            else:
+                os.system(f'rm -f {monitoring_failed_downloads_file}')
```

Comment:
Instead of `os.system`  I would always rather use `subprocess.run`, but in this case you should use the existing function `os.remove`. In case you want to subpress errors if the file does not exist, use the following ([source](https://stackoverflow.com/questions/10840533/most-pythonic-way-to-delete-a-file-which-may-not-exist)) 
```suggestion
                try:
                    os.remove(monitoring_failed_downloads_file)
                except OSError as e:
                    if e.errno != errno.ENOENT: # errno.ENOENT = no such file or directory
                        raise # re-raise exception if a different error occurred
```
- [ ] replace `os.system("rm -f ...")` by `os.remove(...)`

Reply:
fixed